locallib.php

```diff
@@ -443,21 +443,32 @@ function local_boostnavigation_cohort_is_member($userid, $setting) {
     // Initialize variable for memberships.
     static $allmemberships = null;
 
-    // First: If the memberships haven't been fetched yet initially, fetch all of the user's cohort memberships
-    // only once and remember for next calls of this function.
-    if ($allmemberships == null) {
-        // Prepare SQL statement.
+    $cohortversioncache = cache::make('local_boostnavigation', 'cohortversion');
+    $usercohortscache = cache::make('local_boostnavigation', 'usercohorts');
+
+    // Set cohort version cache with current timestamp if the cohort version is not set
+    // from the cohort event observers.
+    if (!$cohortversion = $cohortversioncache->get('cohortversion')) {
+        $cohortversion = time();
```

Comment:
I think refactor out this time() by having a new method get_cohort_revision_cache which internally does the time() call.

Then you can update update_cohort_revision_cache to optionally take an overtime timestamp which you can then set in the unit tests